pkgs/by-name/su/surfer/package.nix

```diff
@@ -0,0 +1,71 @@
+{
+  lib,
+  fetchFromGitLab,
+  rustPlatform,
+  pkg-config,
+  openssl,
+  wayland,
+  autoPatchelfHook,
+  libxkbcommon,
+  libGL,
+  libX11,
+  libXcursor,
+  libXi,
+  stdenv,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "surfer";
+  version = "0.2.0";
+
+  src = fetchFromGitLab {
+    owner = "surfer-project";
+    repo = "surfer";
+    rev = "v${version}";
+    hash = "sha256-C5jyWLs7fdEn2oW5BORZYazQwjXNxf8ketYFwlVkHpA";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    autoPatchelfHook
+  ];
+
+  buildInputs = [
+    openssl
+    wayland
+    libxkbcommon
+    libGL
+    stdenv.cc.cc.libgcc
+  ];
+
+  runtimeDependencies = [
+    openssl
+    wayland
+    libxkbcommon
+    libGL
```

Comment:
```suggestion
```
as these are already in buildInputs, they can be used autoPatchelfHook where necessary. i don't think we need to forcefully add these to the rpath of all binaries

Reply:
When I remove `wayland` and `libxkbcommon` from runtimeDependencies I get
```
event_loop/mod.rs", error: WaylandError(Connection(NoWaylandLib))
```
Alternative is to set
 `          LD_LIBRARY_PATH = "${lib.makeLibraryPath buildInputs}";`
 which is worse imo.

Reply:
winit uses `dlopen` to open X and Wayland library, for better or for worse. Therefore, `runtimeDependencies` is required:

> You can also specify a `runtimeDependencies` variable which lists dependencies to be unconditionally added to rpath of all executables. This is useful for programs that use `dlopen(3)` to load libraries at runtime.

https://nixos.org/manual/nixpkgs/stable/#setup-hook-autopatchelfhook

Reply:
> winit uses dlopen to open X and Wayland libraries and libGL, for better or for worse.

yeah, this is what i assumed when hakan reported the error. you can disregard this

though i do wonder if we need them in builldInputs as well if that's the case? maybe since they're just using dlopen, we don't need to resolve with the pkg-config

Reply:
@getchoo What can I do to resolve this conversation? I have added a comment explaining the dlopen situation. Should I also try to remove pkg-config dependency? I am not familiar with details of linking/finding libs on nix so a more direct request would be helpful.

Reply:
this can be resolved now. please see my newer comments